src/Compatibility/Microsoft.DotNet.ApiCompatibility/AssemblySymbolLoader.cs

```diff
@@ -0,0 +1,299 @@
+            {
+                assemblyName = $"AssemblyLoader_{DateTime.Now:MM_dd_yy_HH_mm_ss_FFF}";
+            }
+
+            _loadedAssemblies = new Dictionary<string, MetadataReference>();
+            var compilationOptions = new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary, nullableContextOptions: NullableContextOptions.Enable);
+            _cSharpCompilation = CSharpCompilation.Create(assemblyName, options: compilationOptions);
+            _resolveReferences = resolveAssemblyReferences;
+        }
+
+        public void AddReferenceSearchDirectories(string paths) => AddReferenceSearchDirectories(SplitPaths(paths));
+
+        public void AddReferenceSearchDirectories(IEnumerable<string> paths)
+        {
+            if (paths == null)
+            {
+                throw new ArgumentNullException(nameof(paths));
+            }
+
+            foreach (string path in paths)
+                _referenceSearchPaths.Add(path);
+        }
+
+        public bool HasRoslynDiagnostics(out IEnumerable<Diagnostic> diagnostics)
+        {
+            diagnostics = _cSharpCompilation.GetDiagnostics();
+            return diagnostics.Any();
+        }
+
+        public bool HasLoadWarnings(out IEnumerable<string> warnings)
+        {
+            warnings = _warnings;
+            return _warnings.Count > 0;
+        }
+
+        private static string[] SplitPaths(string paths) =>
+            paths == null ? null : paths.Split(new char[] { ',', ';' }, StringSplitOptions.RemoveEmptyEntries);
+
+        public IEnumerable<IAssemblySymbol> LoadAssemblies(string paths) => LoadAssemblies(SplitPaths(paths));
+
+        public IEnumerable<IAssemblySymbol> LoadAssemblies(IEnumerable<string> paths)
+        {
+            if (paths == null)
+            {
+                throw new ArgumentNullException(nameof(paths));
+            }
+
+            IEnumerable<MetadataReference> assembliesToReturn = LoadFromPaths(paths);
+
+            List<IAssemblySymbol> result = new List<IAssemblySymbol>();
+            foreach (MetadataReference metadataReference in assembliesToReturn)
+            {
+                ISymbol symbol = _cSharpCompilation.GetAssemblyOrModuleSymbol(metadataReference);
+                if (symbol is IAssemblySymbol assemblySymbol)
+                {
+                    result.Add(assemblySymbol);
+                }
+            }
+
+            return result;
+        }
+
+        public IAssemblySymbol LoadAssembly(string path)
+        {
+            if (path == null)
+            {
+                throw new ArgumentNullException(nameof(path));
+            }
+
+            MetadataReference metadataReference = CreateOrGetMetadataReferenceFromPath(path);
+            return (IAssemblySymbol)_cSharpCompilation.GetAssemblyOrModuleSymbol(metadataReference);
+        }
+
+        public IAssemblySymbol LoadAssembly(string name, Stream stream)
+        {
+            if (name == null)
```

Comment:
how is this name different from the one in the ctor ?

Reply:
The one in the constructor is the assembly name we are creating for the csharp compilation (that is useful when you want to get an IAssemblySymbol from source, because we add the sources to the compilation and return that assembly). However, when loading an assembly from stream or filepath, we add it as a reference to the compilation and return that reference (not the assembly created for the compilation). 

If that is confusing, maybe I can try to figure out a way to change the name of the assembly when doing it from source.

Reply:
I fixed this by adding a parameter to the API that loads an assembly from source files instead.